types/src/network_address/mod.rs

```diff
@@ -183,6 +183,21 @@ pub enum ParseError {
 
     #[error("bcs error: {0}")]
     BCSError(#[from] bcs::Error),
+
+    #[error("NetworkAddress Must start with either Ip4/Ip6/Dns/Dns4/Dns6")]
```

Comment:
"must start with one of" instead of "must start with either".  Minor english thing, where `either` is usually between a pair of options.  Then I would end the sentence with protocol.